inspire_mitmproxy/services/management_service.py

```diff
@@ -83,14 +86,17 @@ def process_request(self, request: MITMRequest) -> MITMResponse:
 
         raise RequestNotHandledInService(self.name, request)
 
-    def get_services(self) -> dict:
-        return {
-            idx: {
-                'class': service.name,
-                'hosts_list': service.hosts_list,
-            }
-            for idx, service in enumerate([cast(BaseService, self)] + self.services)
-        }
+    def get_services(self) -> list:
+        return self.services.to_list()
+
+    def set_services(self, request: MITMRequest) -> list:
+        try:
+            new_services = json_loads(request.body)
+            self.services.replace_from_descrition(new_services)
+            self.services.prepend(self)
```

Comment:
is the mgmt_service already in  `self.services`?

Reply:
No. It's a bit inconsistent, but we agreed that it's the best way: so when you post the list of services you don't specify the manager, it adds itself to the front of the queue always. `replace_from_description` just instantiates the services described (so no manager) and you have to add a manager to it. This is to preserve the manager object, as it also holds configuration and it's not as easily "replaced".